src/main/kotlin/org/dashevo/dpp/identity/IdentityPublicKey.kt

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2020-present, Dash Core Team
+ *
+ * This source code is licensed under the MIT license found in the
+ * COPYING file in the root directory of this source tree.
+ */
+package org.dashevo.dpp.identity
+
+import org.dashevo.dpp.BaseObject
+
+class IdentityPublicKey(var id: Int,
+                        var type: TYPES,
+                        var data: String,
+                        var isEnabled: Boolean) : BaseObject() {
+
+    enum class TYPES(val type: Int) {
+        ECDSA_SECP256K1(1);
+
+        companion object {
+            private val values = values()
+            fun getByCode(code: Int): TYPES {
+                return values.filter { it.type == code }[0]
+            }
+        }
+    }
+
+    constructor(rawIdentityPublicKey: Map<String, Any>) : this(rawIdentityPublicKey["id"] as Int,
+            if (rawIdentityPublicKey["type"] is Int) TYPES.getByCode(rawIdentityPublicKey["type"] as Int) else rawIdentityPublicKey["type"] as TYPES, rawIdentityPublicKey["data"] as String,
```

Comment:
This is section could be easier to read if we use the format:
```
if (...) {
    ...
} else {
    ...
}
```

Reply:
Agreed

Reply:
This is fixed here:  6150374

The solution is to only store the numerical value in any json map for enums.  Therefore we don't need to check the type of the value with an if/else